pkg/pillar/cmd/zedagent/localinfo.go

```diff
@@ -160,9 +267,80 @@ func prepareLocalInfo(ctx *getconfigContext) *profile.LocalAppInfoList {
 		zinfoAppInst.Name = ais.DisplayName
 		zinfoAppInst.Err = encodeErrorInfo(ais.ErrorAndTimeWithSource.ErrorDescription)
 		zinfoAppInst.State = ais.State.ZSwState()
+		ctx.localAppCommandsLock.Lock()
+		for _, appCmd := range ctx.localAppCommands.Cmds {
+			if appCmd.AppUUID == ais.UUIDandVersion.UUID {
+				zinfoAppInst.LastCmdTimestamp = appCmd.LastCompletedTimestamp
+				break
+			}
+		}
+		ctx.localAppCommandsLock.Unlock()
 		msg.AppsInfo = append(msg.AppsInfo, zinfoAppInst)
 		return true
 	}
 	ctx.subAppInstanceStatus.Iterate(addAppInstanceFunc)
 	return &msg
 }
+
+func findAppInstance(
+	ctx *getconfigContext, appUUID uuid.UUID, displayName string) (appInst *types.AppInstanceStatus) {
+	matchApp := func(_ string, value interface{}) bool {
+		ais := value.(types.AppInstanceStatus)
+		if (appUUID == nilUUID || appUUID == ais.UUIDandVersion.UUID) &&
+			(displayName == "" || displayName == ais.DisplayName) {
+			appInst = &ais
+			// stop iteration
+			return false
+		}
+		return true
+	}
+	ctx.subAppInstanceStatus.Iterate(matchApp)
+	return appInst
+}
+
+func readSavedAppCommands(ctx *getconfigContext) (types.LocalAppCommands, error) {
+	appCommands := types.LocalAppCommands{}
+	contents, ts, err := readSavedConfig(
+		ctx.zedagentCtx.globalConfig.GlobalValueInt(types.StaleConfigTime),
```

Comment:
Yes, to answer my own question you are using the same default 5 minutes.

Reply:
It is 10 minutes by default actually.
```
| timer.use.config.checkpoint | integer in seconds | 600 | use checkpointed config if no cloud connectivity |
```